src/burgers1d/main.cpp

```diff
@@ -0,0 +1,244 @@
+using std::endl;
+using std::min;
+
+int main(int argc, char **argv) {
+    using namespace blitzdg;
+    // Physical parameters
+    const real_type xmin =-5.0;
+    const real_type xmax = 5.0;
+
+    const real_type alpha = 1.0;
+    const real_type nu = 0.1;
+    const real_type c = 0.5;
+
+    const real_type finalTime = 1.0;
+    real_type t = 0.0;
+
+    // Numerical parameters:
+    const index_type N = 6;
+    const index_type K = 40;
+    const real_type CFL = 0.75;
+
+    // Build dependencies.
+    Nodes1DProvisioner nodes1DProvisioner(N, K, xmin, xmax);
+    CsvOutputter outputter;
+
+    // Pre-processing. Build grid, and get data we need for initialization.
+    nodes1DProvisioner.buildNodes();
+    nodes1DProvisioner.computeJacobian();
+
+    index_type Np = nodes1DProvisioner.get_NumLocalPoints();
+
+    const real_matrix_type & x = nodes1DProvisioner.get_xGrid();
+
+    real_type min_dx = x(1,0) - x(0,0);
+
+    real_type dt = CFL*min(min_dx/abs(c), min_dx*min_dx/sqrt(nu));
+
+    real_matrix_type u(Np, K);
+    real_matrix_type RHS(Np, K);
+    real_matrix_type resRK(Np, K);
+    real_matrix_type uexact(Np, K);
+
+    firstIndex ii;
+    secondIndex jj;
+
+    printDisclaimer();
+
+    // Intialize u field at t=0
+    burgers1d::Burgers2(u, x, t, alpha, nu, c);
+
+    resRK = 0;
+
+    index_type count = 0;
+
+    const char delim = ' ';
+    outputter.writeFieldToFile("x.dat", x, delim);
+
+    while (t < finalTime) {
+        if ((count % 10) == 0) {
+            string fileName = outputter.generateFileName("u", count);
+            outputter.writeFieldToFile(fileName, u, delim);
+        }
+
+        for (index_type i=0; i < LSERK4::numStages;  i++ ) {
+
+            // Calculate Right-hand side.
+            burgers1d::computeRHS(u, x, t, c, alpha, nu, nodes1DProvisioner, RHS);
+
+            // Compute Runge-Kutta Residual.
+            resRK = LSERK4::rk4a[i]*resRK + dt*RHS;
+
+            // Update solution.
+            u += LSERK4::rk4b[i]*resRK;
+        }
+
+        real_type u_max = normMax(u);
+        if ( u_max > 1e8  || std::isnan(u_max) ) {
+            throw std::runtime_error("A numerical instability has occurred!");
+        }
+
+        t += dt;
+        count++;
+    }
+
+    burgers1d::Burgers2(uexact, x, t, alpha, nu, c);
+    u -= uexact;
+    double err = normMax(u);
+    cout << "Error: " << err << endl;
+
+    return 0;
+} // end main
+
+namespace blitzdg {
+namespace burgers1d {
+real_type Burgers2(real_type x, real_type t, real_type alpha, real_type nu, real_type c) {
+    // Eq (2) from https://www.hindawi.com/journals/mpe/2015/414808/
+    return (c/alpha) - (c/alpha)*tanh( 0.5*(c/nu) * (x - c*t) );
+}
+void Burgers2(real_matrix_type & u, const real_matrix_type & x, real_type t, real_type alpha, real_type nu, real_type c) {
+    for (index_type i=0; i < u.extent(0); i++) {
+        for (index_type j=0; j <u.extent(1); j++) {
+            u(i,j) = Burgers2(x(i,j), t, alpha, nu, c);
```

Comment:
I'd rather see this done with blitz expressions than see the double loop.

Reply:
Agreed that it would be better. I tried the block below, it doesn't compile:

            firstIndex ii;
            secondIndex jj;
            u(ii,jj) = Burgers2(x(ii,jj), t, alpha, nu, c);

Reply:
Fair enough, fixed